Backend/Core/Services/AuthenticationService.cs

```diff
@@ -161,6 +162,17 @@ public async Task ResetPasswordAsync(ResetPasswordDTO resetPasswordDTO)
                         ErrorMessages.NewInfoSamePrevious,
                         HttpStatusCode.BadRequest);
             }
+            if (!await this._userManager.VerifyUserTokenAsync(
+                user, 
+                _userManager.Options.Tokens.PasswordResetTokenProvider, 
+                "ResetPassword", 
+                resetPasswordDTO.Token))
+            {
+
+                throw new HttpException(
+                        "Token Invalid",
+                        HttpStatusCode.BadRequest);
+            }
```

Comment:
We did u add this? The ResetPasswordAsync method'll throw this error

Reply:
The ResetPasswordAsync method does not throw an error if the token has not been used